src/lib/draw/route/route_tool.ts

```diff
@@ -343,9 +334,10 @@ export class RouteTool {
   }
 
   private redraw() {
-    (this.map.getSource(source) as GeoJSONSource).setData(
-      JSON.parse(this.inner.renderGeojson())
-    );
+    let gj = JSON.parse(this.inner.renderGeojson());
+    (this.map.getSource(source) as GeoJSONSource).setData(gj);
+    this.map.getCanvas().style.cursor = gj.cursor;
+    this.snapMode.set(gj.snap_mode);
```

Comment:
Since we're in a .ts file, we can't change  a variable and have the Controls.svelte update easily. Using a store to get around that currently.